assets/js/translator.js

```diff
@@ -65,13 +65,18 @@ if(modeEnabled('translation')) {
         }
 
         function setupWebpageTranslation() {
+            if(window.location.href.indexOf('&qP=') != -1) {
```

Comment:
Doing it this way is really hacky and prone to error. We have `getURLParam` and `HASH_URL_MAP`  for a reason.

Reply:
@sushain97 . I have a question here in the following snippet:

`if(!originalText && ignoreIfEmpty) {`

Would it be a right behaviour to not at all make the text translate request even when only originalText is empty. ie the condition being `if(!originalText) { return; }'` ? translateText() gets called from a  lot of methods and having this check in the method itself wouldn't require `ignoreIfEmpty`.

Reply:
> Would it be a right behaviour to not at all make the text translate request even when only originalText is empty

No. Then if a user clears out the original text field, the result would not change. If we've made the other modes behave like that, it's buggy and should be fixed. APy should be fixed to respond to blank requests properly.

Reply:
Oh, yes! I shall update this accordingly :D